javascript/ql/test/library-tests/TaintTracking/sanitizer-guards.js

```diff
@@ -52,3 +52,14 @@ function reflective() {
     sink(x); // OK
   }
 }
+
+function phi() {
```

Comment:
Can we add this test case as well?
```js
function phi2() {
  let x = source();

  if (somethingThatProbablyMakesItSafe(x) || isSafe(x)) {
    // this input to the phi node for 'x' should be sanitized, but we can not prove that yet
  } else {
    x = null;
  }
  sink(x); // OK, but still flagged
}
```

Reply:
Is `somethingThatProbablyMakesItSafe` supposed to be added as a guard?

The comment is wrong when it says "we can't prove this **yet**" if that's not actually a guard.

Reply:
The main point of the new test case is to ensure that we have a negative test for the dominance part of your changes.

`somethingThatProbablyMakesItSafe` is supposed to be a sanitizing call that we for one reason or another do not detect as such (missing dataflow for the definition of `somethingThatProbablyMakesItSafe`, missing recognition of `somethingThatProbablyMakesItSafe` as a sanitizing function, ...).

Reply:
👍 Good point about adding a negative case.